Framework/API/src/ExperimentInfo.cpp

```diff
@@ -302,31 +304,17 @@ void ExperimentInfo::setInstrument(const Instrument_const_sptr &instr) {
   }
   const auto parInstrument = Geometry::ParComponentFactory::createInstrument(
       sptr_instrument, m_parmap);
+
+  m_infoVisitor = makeOrRetrieveVisitor(*sptr_instrument);
```

Comment:
@OwenArnold Looking at this again: Why does the visitor ever need to live outside the `Instrument`? Couldn't it simply be created internally in `Instrument`? That would avoid messing around with the visitor in `ExperimentInfo` (apart from passing a reference to the `ComponentInfo` factory function. That might make the code in `ExperimentInfo` a bit easier to understand.